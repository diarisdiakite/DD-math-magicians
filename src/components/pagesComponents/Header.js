import React from 'react';
import '../../assets/css/navBar.css';
import { NavLink } from 'react-router-dom';

function Header() {
  return (
    <div className="header">
      <h1 className="title">Math Magicians</h1>
      <nav>
        <ul className="nav-container">
          <li className="nav-item">
            <NavLink to="/" className="nav-link">Home</NavLink>
          </li>
          <li className="nav-item">
            <NavLink to="/calculator" className="nav-link">Calculator</NavLink>
          </li>
          <li className="nav-item">
            <NavLink to="/quote" className="nav-link">Quote</NavLink>
          </li>
        </ul>
      </nav>
    </div>
  );
}

export default Header;
