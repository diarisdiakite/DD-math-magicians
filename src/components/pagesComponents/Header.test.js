import renderer from 'react-test-renderer';
import { MemoryRouter } from 'react-router-dom';
import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import Header from './Header';

describe('Header component', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(
        <MemoryRouter>
          <Header />
        </MemoryRouter>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  describe('Cliking on a links should open the page it links to', () => {
    render(
      <MemoryRouter>
        <Header />
      </MemoryRouter>,
    );
    const homeNavLink = screen.getByRole('link', { name: 'Home' });
    describe('Should led the user to the Home page when he clicks Home navelink', () => {
      expect(homeNavLink).toBeDefined();

      const expectedPath = '/';
      userEvent.click(homeNavLink);
      expect(window.location.pathname).toEqual(expectedPath);
    });
  });
});
