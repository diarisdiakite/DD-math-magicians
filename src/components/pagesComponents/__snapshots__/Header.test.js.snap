// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component renders correctly 1`] = `
<div
  className="header"
>
  <h1
    className="title"
  >
    Math Magicians
  </h1>
  <nav>
    <ul
      className="nav-container"
    >
      <li
        className="nav-item"
      >
        <a
          aria-current="page"
          className="nav-link active"
          href="/"
          onClick={[Function]}
        >
          Home
        </a>
      </li>
      <li
        className="nav-item"
      >
        <a
          className="nav-link"
          href="/calculator"
          onClick={[Function]}
        >
          Calculator
        </a>
      </li>
      <li
        className="nav-item"
      >
        <a
          className="nav-link"
          href="/quote"
          onClick={[Function]}
        >
          Quote
        </a>
      </li>
    </ul>
  </nav>
</div>
`;
